File: coding-frontend/src/app/_models/Post.ts
import { Tag } from './Tag';
import { Commentary } from './Comments';

export interface Post {
    id: string,
    title: string,
    description: string,
    text: string,
    createdat: Date,
    publishedat: Date,
    tags?: Tag[]
    comments?: Commentary[]
    readingtime: number,
    userid: number,
    photourl?: string,
    dateCreated?: string,
    readingTime?: string
}

File: coding.API/Models/Posts/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using coding.API.Models.PostTags;
using coding.API.Models.Posts.Comments;
using coding.API.Models.Photos;

namespace coding.API.Models.Posts
{
    public class Post: BaseEntity
    {
        [Required]
        public string Title { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public string Text { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? PublishedAt { get; set; }
        public int ReadingTime { get; set; } // In minutes
        public List<PostTag> PostTags { get; set; }
        public List<PostPhoto> Photos { get; set; }
        public Guid UserId { get; set; }       
        public List<Comment> Comments { get; set; }
    }
} 